src/Detail/Normalization/JMSSerializer/Handler/DateImmutableHandler.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Detail\Normalization\JMSSerializer\Handler;
+
+use DateTime;
+use DateTimeImmutable;
+use DateTimeZone;
+
+use JMS\Serializer\Context;
+use JMS\Serializer\GraphNavigator;
+use JMS\Serializer\Handler\DateHandler as BaseDateHandler;
+use JMS\Serializer\JsonDeserializationVisitor;
+use JMS\Serializer\VisitorInterface;
+use JMS\Serializer\XmlDeserializationVisitor;
+
+use Detail\Normalization\Exception;
+use Detail\Normalization\JMSSerializer\PhpDeserializationVisitor;
+
+class DateImmutableHandler extends BaseDateHandler
+{
+    use DatePhpDeserializationTrait;
+
+    /**
+     * @param string $defaultFormat
+     * @param string $defaultTimezone
+     */
+    public function __construct($defaultFormat = DateTime::ISO8601, $defaultTimezone = 'UTC')
+    {
+        $this->format = $defaultFormat;
+        $this->timezone = new DateTimeZone($defaultTimezone);
+
+        parent::__construct($defaultFormat, $defaultTimezone);
```

Comment:
Parent defines those variable as private, therefore are not available in local context